cxx/structureTranslationModule/agent/StructureTranslationAgent.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
+ 
+#include "sc-agents-common/utils/CommonUtils.hpp"
+#include "sc-agents-common/utils/AgentUtils.hpp"
+#include "sc-agents-common/utils/IteratorUtils.hpp"
+#include <iterator>
+#include "constants/TranslationConstants.hpp"
+
+#include "keynodes/TranslationKeynodes.hpp"
+
+#include "StructureTranslationAgent.hpp"
+
+#include<string>
+
+namespace structureTranslationModule
+{
+
+SC_AGENT_IMPLEMENTATION(StructureTranslationAgent)
+{
+  ScAddr const & actionNode = otherAddr;
+  try
+  {
+    if (checkActionClass(actionNode) == SC_FALSE)
+      return SC_RESULT_OK;
+    SC_LOG_INFO("StructureTranslationAgent started");
+
+    ScAddr const & structuresSet = utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, actionNode, scAgentsCommon::CoreKeynodes::rrel_1);
+
+    std::string translation;
+
+    auto const & structIterator = m_memoryCtx.Iterator3(structuresSet, ScType::EdgeAccessConstPosPerm, ScType::NodeConstStruct);
```

Comment:
```suggestion
    ScIterator3Ptr const & structIterator = m_memoryCtx.Iterator3(structuresSet, ScType::EdgeAccessConstPosPerm, ScType::NodeConstStruct);
```
We use auto keyword only with difficult long types